cmd/topicctl/subcmd/get.go

```diff
@@ -38,109 +32,240 @@ type getCmdConfig struct {
+
+			adminClient, err := getConfig.shared.getAdminClient(ctx, sess, true)
+			if err != nil {
+				return err
+			}
+			defer adminClient.Close()
 
-	adminClient, err := getConfig.shared.getAdminClient(ctx, sess, true)
-	if err != nil {
-		return err
+			cliRunner := cli.NewCLIRunner(adminClient, log.Infof, !noSpinner)
+			return cliRunner.GetBrokers(ctx, getConfig.full)
+		},
 	}
-	defer adminClient.Close()
-
-	cliRunner := cli.NewCLIRunner(adminClient, log.Infof, !noSpinner)
-
-	resource := args[0]
-
-	switch resource {
-	case "balance":
-		var topicName string
-
-		if len(args) == 2 {
-			topicName = args[1]
-		} else if len(args) > 2 {
-			return fmt.Errorf("Can provide at most one positional argument with brokers")
-		}
-
-		return cliRunner.GetBrokerBalance(ctx, topicName)
-	case "brokers":
-		if len(args) > 1 {
-			return fmt.Errorf("Can only provide one positional argument with brokers")
-		}
-
-		return cliRunner.GetBrokers(ctx, getConfig.full)
-	case "config":
-		if len(args) != 2 {
-			return fmt.Errorf("Must provide broker ID or topic name as second positional argument")
-		}
-
-		return cliRunner.GetConfig(ctx, args[1])
-	case "groups":
-		if len(args) > 1 {
-			return fmt.Errorf("Can only provide one positional argument with groups")
-		}
-
-		return cliRunner.GetGroups(ctx)
-	case "lags":
-		if len(args) != 3 {
-			return fmt.Errorf("Must provide topic and groupID as additional positional arguments")
-		}
-
-		return cliRunner.GetMemberLags(
-			ctx,
-			args[1],
-			args[2],
-			getConfig.full,
-			getConfig.sortValues,
-		)
-	case "members":
-		if len(args) != 2 {
-			return fmt.Errorf("Must provide group ID as second positional argument")
-		}
-
-		return cliRunner.GetGroupMembers(ctx, args[1], getConfig.full)
-	case "partitions":
-		if len(args) != 2 {
-			return fmt.Errorf("Must provide topic as second positional argument")
-		}
-		topicName := args[1]
-
-		return cliRunner.GetPartitions(ctx, topicName)
-	case "offsets":
-		if len(args) != 2 {
-			return fmt.Errorf("Must provide topic as second positional argument")
-		}
-		topicName := args[1]
-
-		return cliRunner.GetOffsets(ctx, topicName)
-	case "topics":
-		if len(args) > 1 {
-			return fmt.Errorf("Can only provide one positional argument with args")
-		}
-
-		return cliRunner.GetTopics(ctx, getConfig.full)
-	default:
-		return fmt.Errorf("Unrecognized resource type: %s", resource)
+}
+
+func configCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "config [broker or topic]",
+		Short: "Displays configuration for the provider broker or topic.",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			sess := session.Must(session.NewSession())
+
+			adminClient, err := getConfig.shared.getAdminClient(ctx, sess, true)
+			if err != nil {
+				return err
+			}
+			defer adminClient.Close()
+
+			cliRunner := cli.NewCLIRunner(adminClient, log.Infof, !noSpinner)
```

Comment:
This block looks like it ends up being boilerplate for all the subcmd functions. Any way to reduce its redundancy?